apps/web/src/App.tsx

```diff
@@ -237,8 +238,30 @@ function jwtHasKey(key: string) {
 }
 
 function RequiredAuth({ children }: any) {
+  const searchParams = useSearchParams();
   const { logout } = useContext(AuthContext);
   const location = useLocation();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const id = localStorage.getItem('blueprintId');
+    const token = getToken();
+
+    if (id && token) {
+      navigate(`/templates/create?page=${ActivePageEnum.WORKFLOW}`, {
+        replace: true,
+      });
+    }
+  }, [localStorage.getItem('blueprintId'), getToken()]);
+
+  useEffect(() => {
+    if (searchParams.blueprintId) {
+      localStorage.setItem('blueprintId', searchParams.blueprintId);
+      navigate(`/templates/create?page=${ActivePageEnum.WORKFLOW}`, {
+        replace: true,
+      });
+    }
+  }, [searchParams.blueprintId]);
```

Comment:
I have no full context why we need to set the actions for the blueprint inside of the functionality that handles the authentication. I am having the feeling it is bloating this function and making the logic with more branches, but right now I have no more arguments to back that up.
Also as `RequiredAuth` is being applied in many routes. Do we want to apply this logic in all of those routes?

Reply:
"Do we want to apply this logic in all of those routes?" this was the reason why I put it here I like to redirect them to the blueprint modal from anywhere they land on our web app. Maybe I should wrap it in a hook?

Reply:
If that's the goal, wrap it in a hook would help a lot, indeed. 👍🏻

Reply:
Moved to a hook now